File: Count Subarrays Where Max Element Appears at Least K Times.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k) 
{
    const n = nums.length;
    let max = Math.max(...nums);
    let start = 0;
    let count = 0;

    for(let i = 0; i < n; i++)
    {
        if(nums[i] === max)
        {
            k--;
        }
        while(k == 0)
        {
            if(nums[start] == max)
            {
                k++;
            }
            start++;
        }
        count += start;
    }
    return count;
};

File: Count Subarrays Where Max Element Appears at Least K Times.py
class Solution(object):
    def countSubarrays(self, nums, k):
        
        Max = max(nums)
        start = count = 0

        for i in range(len(nums)):
            if nums[i] == Max:
                k -= 1
            
            while k == 0:
                if nums[start] == Max:
                    k += 1
                start += 1
            
            count += start
        
        return count
        